### PR TITLE
Extra sound effects

### PR DESCRIPTION
**Pull request by MirceaKitsune**
This adds various sound effects to minetest_game, including:

- Open and close sounds for doors, taken from the Home Decor mod by VanessaE. A sound file for gates is also included, to be used with fence gates when such is added.

- Water and lava sounds. They are looped on any liquid source that causes flowing liquid (most noticeable with non-finite fluids). Note that they're only triggered for sources that don't touch other sources, otherwise a leaking lake bombards the server and the player's headphones with noises.

- Interaction sounds for chests and furnaces. A chest sound is played when opening a chest, while the furnace sound is triggered when a furnace becomes active. A sound for the workbench is also included, to be used once a workbench is included.

- Eating sound, triggered when using an apple or bread.

Please test, post your opinion, and merge when everyone's ready. Water sounds should be tested specifically, to see if they behave correctly in all circumstances. Original sound sources (note that most were modified):

http://freesound.org/people/Slanesh/sounds/31768/
http://freesound.org/people/j1987/sounds/106116/
http://www.pdsounds.org/sounds/water_over_rocks_as_tide_recedes
http://www.pdsounds.org/sounds/squeaky_door
http://www.pdsounds.org/sounds/rusty_metal_door_clasp
http://www.pdsounds.org/sounds/wood_and_cutlery
http://www.pdsounds.org/sounds/crunching_munching_grinding

### CODE REVIEW

**Comment by paramat** [comment]
@MirceaKitsune the liquid sounds look very heavy, please can you update this PR if you still want to submit the simpler sounds?

File: mods/default/functions.lua [hunk 1]
```diff
@@ -84,6 +84,78 @@ function default.node_sound_glass_defaults(table)
...
+end)
+
+minetest.register_on_dignode(function(pos, _, _)
+	flow_sound_stop(pos)
+end)
+
+minetest.register_abm({
+	nodenames = {"default:water_source"},
+	neighbors = {"default:water_flowing"},
+	interval = 0.5,
```
Comment (HybridDog) [inline mods/default/functions.lua]: as far as l know the minimum interval is 1

**Comment by MirceaKitsune** [comment]
@paramat What do you mean by heavy? Should I increase their pitch, or are you referring to the file size? To me they sound very good as they are, but if other users want I can try tweaking them more I guess, next time I work on Minetest.

**Comment by Ferk** [comment]
I think he means heavy in terms of processing time required.

**Comment by paramat** [comment]
Yes i mean a heavy processing load.

**Comment by MirceaKitsune** [comment]
Sounds like too many things need to be changed, and most of the sounds removed. Perhaps it's best to abandon this pull and re-add some of these features more slowly some other time. The flow (looped liquid source) sounds need a better system to prevent spam anyway, could even require engine changes.

**Comment by HybridDog** [comment]
>Eat sound is too crunchy, too slow and too long.

How about the ones of hungry games? https://github.com/Splizard/hungry_games/tree/master/mods/survival_modpack/survival_hunger/sounds

**Pull request closed without merging**